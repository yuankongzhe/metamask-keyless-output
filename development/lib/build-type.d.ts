import { Infer, Struct } from 'superstruct';

export type Unique<Element extends Struct<any>> = (
  struct: Struct<Infer<Element>[], Infer<Element>>,
  eq?: (a: Infer<Element>, b: Infer<Element>) => boolean,
) => Struct<Infer<Element>[], Infer<Element>>;
