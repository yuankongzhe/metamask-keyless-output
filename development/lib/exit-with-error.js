/**
 * Exit the process with an error message.
 *
 * Note that this should be called before the process ends, but it will not
 * itself end the process. This is because the Node.js documentation strongly
 * advises against calling `process.exit` directly.
 *
 * @param {string} errorMessage - The error message that is causing the non-
 *   zero exit code.
 */
function exitWithError(errorMessage) {
  console.error(errorMessage);
  process.exitCode = 1;
  if (process.env.CI) {
    // For some reason, the CI environment does not exit when process.exitCode
    // is set, or at least doesn't terminate immediately. Something may be
    // keeping the event loop from ending. The result is that sometimes jobs
    // fail because of timeout even though none of the individual tests fail.
    // This is a workaround.
    process.exit();
  }
}

module.exports = { exitWithError };
