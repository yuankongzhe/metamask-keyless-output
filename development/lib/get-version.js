const { version: manifestVersion } = require('../../package.json');
const { loadBuildTypesConfig } = require('./build-type');

/**
 * Get the current version of the MetaMask extension. The base manifest version
 * is modified according to the build type and version.
 *
 * The build version is needed because certain build types (such as beta) may
 * be released multiple times during the release process.
 *
 * @param {string} buildType - The build type.
 * @param {number} buildVersion - The build version.
 * @returns {string} The MetaMask extension version.
 */
function getVersion(buildType, buildVersion) {
  return loadBuildTypesConfig().buildTypes[buildType].isPrerelease === true
    ? `${manifestVersion}-${buildType}.${buildVersion}`
    : manifestVersion;
}

module.exports = { getVersion };
