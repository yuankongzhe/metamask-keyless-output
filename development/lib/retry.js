/**
 * Re-runs the given function until it returns a resolved promise or the number
 * of retries is exceeded, whichever comes first (with an optional delay in
 * between retries).
 *
 * @param {object} args - A set of arguments and options.
 * @param {number} args.retries - The maximum number of times to re-run the
 * function on failure.
 * @param {number} [args.delay] - The amount of time (in milliseconds) to wait in
 * between retries. (Default: 0)
 * @param {string} [args.rejectionMessage] - The message for the rejected promise
 * this function will return in the event of failure. (Default: "Retry limit
 * reached")
 * @param {boolean} [args.retryUntilFailure] - Retries until the function fails.
 * @param {Function} functionToRetry - The function that is run and tested for
 * failure.
 * @returns {Promise<* | null | Error>} a promise that either resolves with one of the following:
 * - If successful, resolves with the return value of functionToRetry.
 * - If functionToRetry fails while retryUntilFailure is true, resolves with null.
 * - Otherwise it is rejected with rejectionMessage.
 */
async function retry(
  {
    retries,
    delay = 0,
    rejectionMessage = 'Retry limit reached',
    retryUntilFailure = false,
  },
  functionToRetry,
) {
  let attempts = 0;
  while (attempts <= retries) {
    if (attempts > 0 && delay > 0) {
      await new Promise((resolve) => setTimeout(resolve, delay));
    }

    try {
      const result = await functionToRetry();
      if (!retryUntilFailure) {
        return result;
      }
    } catch (error) {
      if (error.message === "Exited with code '1'") {
        console.log("retry() received: Exited with code '1'");
      } else {
        console.error('error caught in retry():', error);
      }

      if (attempts < retries) {
        console.log('Ready to retry() again');
      }

      if (retryUntilFailure) {
        return null;
      }
    } finally {
      attempts += 1;
    }
  }

  throw new Error(rejectionMessage);
}

module.exports = { retry };
