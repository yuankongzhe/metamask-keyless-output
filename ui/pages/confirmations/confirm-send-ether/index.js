export { default } from './confirm-send-ether';
