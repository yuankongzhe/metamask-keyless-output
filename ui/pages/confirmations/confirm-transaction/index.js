import ConfirmTransaction from './confirm-transaction.component';

export default ConfirmTransaction;
