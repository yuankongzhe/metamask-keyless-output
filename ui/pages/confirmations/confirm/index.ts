export { default as Confirm } from './confirm';
