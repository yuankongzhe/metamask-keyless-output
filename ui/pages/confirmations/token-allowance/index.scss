.token-allowance-container {
  &__icon-display-content {
    width: fit-content;
    height: 40px;
    box-sizing: border-box;
    border-radius: 100px;
    position: relative;

    &__siteimage-identicon {
      width: 24px;
      height: 24px;
      box-shadow: none;
      background: none;
    }
  }

  a.token-allowance-container__verify-link {
    width: fit-content;
    margin-inline-start: auto;
    margin-inline-end: auto;
    padding: 0;
  }

  a.token-allowance-container__view-details {
    width: fit-content;
    margin-inline-start: 108px;
    margin-inline-end: 108px;
  }

  &__card-wrapper {
    width: 100%;
  }

  &__data {
    width: 100%;
  }

  &__full-tx-content {
    max-width: 100%;
  }

  &__custom-nonce-warning {
    width: 100%;
    height: 30px;
  }
}
