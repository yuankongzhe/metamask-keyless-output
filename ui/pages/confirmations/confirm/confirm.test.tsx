import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { unapprovedPersonalMsg } from '../../../../test/data/confirmations/personal_sign';
import mockState from '../../../../test/data/mock-state.json';
import { renderWithProvider } from '../../../../test/lib/render-helpers';

import Confirm from './confirm';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    replace: jest.fn(),
  }),
}));

const middleware = [thunk];

describe('Confirm', () => {
  it('should render', () => {
    const mockStore = configureMockStore(middleware)(mockState);
    const { container } = renderWithProvider(<Confirm />, mockStore);
    expect(container).toBeDefined();
  });

  it('should match snapshot for personal signature', async () => {
    const mockStatePersonalSign = {
      ...mockState,
      metamask: {
        ...mockState.metamask,
      },
      confirm: { currentConfirmation: unapprovedPersonalMsg },
    };
    const mockStore = configureMockStore(middleware)(mockStatePersonalSign);
    const { container } = renderWithProvider(<Confirm />, mockStore);
    expect(container).toMatchSnapshot();
  });
});
