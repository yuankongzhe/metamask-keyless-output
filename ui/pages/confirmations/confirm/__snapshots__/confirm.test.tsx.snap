// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm should match snapshot for personal signature 1`] = `
<div>
  <div
    class="mm-box multichain-page mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--width-full mm-box--height-full mm-box--background-color-background-alternative"
  >
    <div
      class="mm-box multichain-page__inner-container mm-box--display-flex mm-box--flex-direction-column mm-box--width-full mm-box--height-full mm-box--background-color-background-default"
    >
      <div
        class="mm-box confirm_header__wrapper mm-box--display-flex mm-box--justify-content-space-between mm-box--align-items-center"
      >
        <div
          class="mm-box mm-box--padding-4 mm-box--display-flex mm-box--align-items-flex-start"
        >
          <div
            class="mm-box mm-box--margin-top-2 mm-box--display-flex"
          >
            <div
              class=""
            >
              <div
                class="identicon"
                style="height: 32px; width: 32px; border-radius: 16px;"
              >
                <div
                  style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 32px; height: 32px; display: inline-block; background: rgb(3, 73, 94);"
                >
                  <svg
                    height="32"
                    width="32"
                    x="0"
                    y="0"
                  >
                    <rect
                      fill="#FB184D"
                      height="32"
                      transform="translate(-0.010232866814926739 -2.467961361206475) rotate(274.3 16 16)"
                      width="32"
                      x="0"
                      y="0"
                    />
                    <rect
                      fill="#1888F2"
                      height="32"
                      transform="translate(0.7019171336752298 13.759175028660584) rotate(203.5 16 16)"
                      width="32"
                      x="0"
                      y="0"
                    />
                    <rect
                      fill="#FA8900"
                      height="32"
                      transform="translate(28.820219003567207 11.830135279564011) rotate(192.7 16 16)"
                      width="32"
                      x="0"
                      y="0"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network confirm_header__avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
            >
              C
            </div>
          </div>
          <div
            class="mm-box mm-box--margin-inline-start-4"
          >
            <p
              class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
            />
            <p
              class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative"
            >
              Chain 5
            </p>
          </div>
        </div>
        <div
          class="mm-box mm-box--padding-4 mm-box--display-flex mm-box--align-items-flex-end"
        >
          <div
            class="mm-box mm-box--display-flex mm-box--justify-content-flex-end"
            style="align-self: flex-end;"
          >
            <div>
              <div
                aria-describedby="tippy-tooltip-2"
                class=""
                data-original-title="Account details"
                data-tooltipped=""
                style="display: inline;"
                tabindex="0"
              >
                <button
                  aria-label="Account details"
                  class="mm-box mm-button-icon mm-button-icon--size-md mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-icon-default mm-box--background-color-transparent mm-box--rounded-lg"
                >
                  <span
                    class="mm-box mm-icon mm-icon--size-md mm-box--display-inline-block mm-box--color-inherit"
                    style="mask-image: url('./images/icons/info.svg');"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mm-box mm-box--margin-bottom-4 mm-box--padding-4 mm-box--display-flex mm-box--width-full mm-box--background-color-primary-muted"
      >
        <span
          class="mm-box mm-icon mm-icon--size-md mm-box--margin-right-2 mm-box--display-inline-block mm-box--color-info-default"
          style="mask-image: url('./images/icons/info.svg');"
        />
        <p
          class="mm-box mm-text mm-text--body-xs mm-box--color-text-default"
        >
          Your selected account (0x0dcd5...3e7bc) is different than the account trying to sign ()
        </p>
      </div>
      <div
        class="mm-box multichain-page-content mm-box--padding-4 mm-box--display-flex mm-box--flex-direction-column mm-box--width-full mm-box--height-full mm-box--background-color-background-alternative"
      >
        <h2
          class="mm-box mm-text mm-text--heading-lg mm-text--text-align-center mm-box--padding-top-4 mm-box--padding-bottom-2 mm-box--color-text-default"
        >
          Signature request
        </h2>
        <p
          class="mm-box mm-text mm-text--body-md mm-text--text-align-center mm-box--padding-bottom-4 mm-box--color-text-alternative"
        >
          Only sign this message if you fully understand the content and trust the requesting site
        </p>
        <div
          class="mm-box mm-box--margin-bottom-4 mm-box--padding-2 mm-box--background-color-background-default mm-box--rounded-md"
        >
          <div
            class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
            style="overflow-wrap: anywhere; min-height: 24px;"
          >
            <div
              class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
            >
              <p
                class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
              >
                Request from
              </p>
              <div>
                <div
                  aria-describedby="tippy-tooltip-3"
                  class=""
                  data-original-title="This is the site asking for your signature."
                  data-tooltipped=""
                  style="display: flex;"
                  tabindex="0"
                >
                  <span
                    class="mm-box mm-icon mm-icon--size-md mm-box--margin-left-1 mm-box--display-inline-block mm-box--color-icon-muted"
                    style="mask-image: url('./images/icons/question.svg');"
                  />
                </div>
              </div>
            </div>
            <div
              class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center"
            >
              <p
                class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
              >
                https://metamask.github.io
              </p>
            </div>
          </div>
        </div>
        <div
          class="mm-box mm-box--margin-bottom-4 mm-box--padding-2 mm-box--background-color-background-default mm-box--rounded-md"
        >
          <div
            class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
            style="overflow-wrap: anywhere; min-height: 24px;"
          >
            <div
              class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
            >
              <p
                class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
              >
                Message
              </p>
            </div>
            <div
              class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center"
            >
              <p
                class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
              >
                Example \`personal_sign\` message
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mm-box multichain-page-footer mm-box--padding-4 mm-box--display-flex mm-box--gap-4 mm-box--width-full"
      >
        <button
          class="mm-box mm-text mm-button-base mm-button-base--size-lg mm-button-base--block mm-button-secondary mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-pill mm-box--border-color-primary-default box--border-style-solid box--border-width-1"
        >
          Cancel
        </button>
        <button
          class="mm-box mm-text mm-button-base mm-button-base--size-lg mm-button-base--block mm-button-primary mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-inverse mm-box--background-color-primary-default mm-box--rounded-pill"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
